editor/modes/visual-editor/block.js

```diff
@@ -11,84 +11,116 @@ import Toolbar from 'components/toolbar';
 import BlockMover from 'components/block-mover';
 import BlockSwitcher from 'components/block-switcher';
 
-function VisualEditorBlock( props ) {
-	const { block } = props;
-	const settings = wp.blocks.getBlockSettings( block.blockType );
+class VisualEditorBlock extends wp.element.Component {
+	constructor() {
+		super( ...arguments );
+		this.bindBlockNode = this.bindBlockNode.bind( this );
+	}
 
-	let BlockEdit;
-	if ( settings ) {
-		BlockEdit = settings.edit || settings.save;
+	bindBlockNode( node ) {
+		this.node = node;
 	}
 
-	if ( ! BlockEdit ) {
-		return null;
+	componentWillReceiveProps( newProps ) {
+		if (
+			this.props.order !== newProps.order &&
+			this.props.isSelected &&
+			newProps.isSelected
+		) {
+			this.previousOffset = this.node.getBoundingClientRect().top;
+		}
 	}
 
-	const { isHovered } = props;
-	const isSelected = props.isSelected;
-	const className = classnames( 'editor-visual-editor__block', {
-		'is-selected': isSelected,
-		'is-hovered': isHovered
-	} );
+	componentDidUpdate() {
+		if ( this.previousOffset ) {
+			window.scrollTo(
+				window.scrollX,
+				window.scrollY + this.node.getBoundingClientRect().top - this.previousOffset
+			);
+			this.previousOffset = false;
```

Comment:
I'm not a fan of using `false` as an empty state. Could we assign to `null`, `undefined`, or even `delete this.previousOffset` instead?